src/app/src/components/EmbeddedMapTextSearchLabelConfig.jsx

```diff
@@ -0,0 +1,92 @@
+import React, { useState, useEffect } from 'react';
+import Typography from '@material-ui/core/Typography';
+import { withStyles } from '@material-ui/core/styles';
+import ControlledTextInput from './ControlledTextInput';
+import { DEFAULT_SEARCH_TEXT } from '../util/constants';
+
+const styles = {
+    subsectionHeader: {
+        color: 'rgb(0, 0, 0)',
+        fontSize: '15px',
+        fontWeight: 'bold',
+        margin: '10px 0',
+    },
+    reset: {
+        background: 'none',
+        border: 'none',
+        padding: '0',
+        cursor: 'pointer',
+        textDecoration: 'underline',
+    },
+};
+
+function EmbeddedMapTextSearchLabelConfig({
+    textSearchLabel,
+    setTextSearchLabel,
+    anyFieldSearchable,
+}) {
+    const [textInputValue, setTextInputValue] = useState(textSearchLabel);
+
+    const saveTextValue = value => {
+        setTextInputValue(value);
+        setTextSearchLabel(value);
+    };
+
+    const setTextOnBlur = () => {
+        if (textInputValue.trim()) {
+            saveTextValue(textInputValue);
+        } else {
+            saveTextValue(DEFAULT_SEARCH_TEXT);
+        }
+    };
+
+    const setTextOnChange = value => {
+        if (value.trim()) {
+            saveTextValue(value);
+        } else {
+            setTextInputValue(value);
+        }
+    };
+    const onReset = e => {
+        e.preventDefault();
+        console.log('reset');
+        saveTextValue(DEFAULT_SEARCH_TEXT);
+    };
+
+    useEffect(() => {
+        if (!anyFieldSearchable) {
+            saveTextValue(DEFAULT_SEARCH_TEXT);
+        }
+    }, [anyFieldSearchable]);
+
+    return (
+        <div>
+            <Typography style={styles.subsectionHeader}>
+                Search box label
+            </Typography>
+            <Typography>
+                Set the text to display on your side bar when fields are marked
+                as searchable.
+                <button
```

Comment:
When this button renders, there is no space before it. This should fix that:

```suggestion
                Set the text to display on your side bar when fields are marked
                as searchable.
                {" "}
                <button
```